docs/ssh.md

```diff
@@ -0,0 +1,114 @@
+# SSH into a VirtualMachineInstance
+
+Every VMI provides a `/tunnel` subresource that can be used to create a websocket backed
+network tunnel with a port inside the instance.
+
+One use-case for this subresource is to forward SSH traffic into the VMI either from the CLI
+or a web-UI.
+
+## Usage
+
+### virtcli
+
+To connect to a VMI from your local machine, `virtcli` can be used as a SSH `ProxyCommand`
+and will forward the SSH connection through the Kubernetes API Server to the targeted VMI.
+
+```sh
+ssh -o 'ProxyCommand=virtcli ssh -runAsProxyCommand=true testvmi' localhost
```

Comment:
That is a nice usage I did not think about before. :+1: 

Note that especially with ssh it is common to do a `VPN`. I would therefore really suggest to just introduce a `port-forward` functionality, instead of limiting things to the ssh port. We don't "prevent" anyone from misuse this way.

Reply:
@kwiesmueller what do you think about the follwoing: Since the vms do not really map to hostnames or domain names and instead map to a `name`, `namespace` pair, what if  we allow to specify VMs on `virtctl ssh --runAsProxyCommand=true` in more advanced ways, similar to `kubectl port-forward svc/<myservice>`? We could allow specifying VMs  in these ways:

 * `virtctl ssh --runAsProxyCommand=true <name> -n <namespace>`
 * `virtctl ssh --runAsProxyCommand=true vmi/<name>.<namespace>` 
 * `virtctl ssh --runAsProxyCommand=true vm/<name>` # uses current namespace
 * `virtctl ssh --runAsProxyCommand=true vmi/<name>`
 * `virtctl ssh --runAsProxyCommand=true vm/<name>.<namespace>` 
 * `virtctl ssh --runAsProxyCommand=true virtualmachineinstance/<name>.<namespace>` 
 * `virtctl ssh --runAsProxyCommand=true virtualmachine/<name>.<namespace>`   
 
 Because if we do that, one can write these `ssh_config` entries:
 
Redirect all `vmi/*` entries through `virtctl`:
 ```
 Host vmi/*
    User root
    ProxyCommand virtctl ssh --runAsProxyCommand=true %h
 ```
 
 One can then naively use `ssh` in a clear understandable way without wrapping: `ssh vmi/myvm.mynamespace`. We may even find a way to also express the context somehow, but I would assume that we are just logged in to the right cluster for now.
 
 
 Treat all vmis in `namespace1` differently and propose to use a different identity file for all VMIs in that namespace:
 
 ```
Host vmi/*.namespace1
    User root
    ProxyCommand virtctl ssh --runAsProxyCommand=true %h
    IdentityFile /non/default/id_rsa
```

Here the same with using `vm/myvm.namespace1` instead of `vmi/myvm.namespace1`

```    
Host vm/*.namespace1
    User root
    ProxyCommand virtctl ssh --runAsProxyCommand=true %h
    IdentityFile /non/default/id_rsa
```

Reply:
I like the idea.
Right now it only works on vmis. Will have to check what's needed for vms, probably not much.
Would vmi be the default? Or what should happen if you do `virtctl ssh --runAsProxyCommand=true <name> -n <namespace>`
Also with the change now it would be `virtctl portforward --forwardToStdout -n <namespace> <name> 22`.